nav2_route/src/goal_intent_extractor.cpp

```diff
@@ -193,6 +213,36 @@ Route GoalIntentExtractor::pruneStartandGoal(
   return pruned_route;
 }
 
+bool GoalIntentExtractor::isNodeValid(unsigned int node_index, 
+                                      const geometry_msgs::msg::PoseStamped & pose)
+{
+  unsigned int s_mx, s_my, g_mx, g_my; 
+  costmap_sub_->getCostmap()->worldToMap(pose.pose.position.x, pose.pose.position.y, 
+                                         s_mx, s_my);
+
+  // double check the frames, probably need to move into the map frame...
+  float goal_x = (*graph_)[node_index].coords.x;
```

Comment:
Don't look up multiple times, also use the api `graph_->at(idx)` which goes around the length checking to increase speed